src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -57,18 +57,28 @@
 (defn wrap-with-environment-validation
   "Middleware function which validates the presence and syntactical content
   of an environment in a ring request.  If validation fails, a :bad-request
-  slingshot exception is thrown."
-  [handler]
-  (fn [request]
-    (let [environment (get-environment-from-request request)]
-      (cond
-        (nil? environment)
-        (ringutils/throw-bad-request!
-         (i18n/tru "An environment parameter must be specified"))
+  slingshot exception is thrown. If the optional-environment-param is
+  provided as true, it allows for the endpoint to have its query param
+  environment to be optional. However if the param is provided, it will
+  still validate that param for the given request."
+  ([handler optional-environment-param]
+   (fn [request]
+     (let [environment (get-environment-from-request request)]
+       (if (and optional-environment-param
```

Comment:
If `optional-environment-param` is specified as `false` and the environment is `nil`, won't this `if` statement evaluate to false and pass the request through to the handler.  That doesn't seem right.

One alternative here might be to just have the 1 argument version have no other logic in it other than to call through to the 2 argument version, which would now have the `cond` in it.  And the first `cond` statement, on detecting a `nil` environment, could throw the error vs. threading through to the handler based on the value of the `optional-environment-param`.  WDYT?